packages/nameguard-js/src/nameguard-js.test.ts

```diff
@@ -34,37 +34,75 @@ describe("NameGuardJS", () => {
     const localNameguard = createClient({
       // not a real endpoint, will error if used
       endpoint: "http://localhost:1234",
-      network: "sepolia",
       publicClient
     });
 
     expect(
       // this should try to fetch from the endpoint
       localNameguard.getSecurePrimaryName(
         "0xb8c2C29ee19D8307cb7255e1Cd9CbDE883A267d5",
-        { network: "mainnet" }
+        { computeNameGuardReport: true }
       )
-    ).rejects.toThrow("fetch failed");
+    ).rejects.toThrow("request to http://localhost:1234/secure-primary-name/mainnet/0xb8c2C29ee19D8307cb7255e1Cd9CbDE883A267d5 failed");
   });
 
-  it("should use the API for requests with a requested nameguard report", () => {
-    const publicClient = createPublicClient({
+  it("should throw an error for network mismatch on mainnet", () => {
+    const sepoliaClient = createPublicClient({
+      chain: sepolia,
+      transport: http(),
+    });
+
+    expect(() => {
+      createClient({
+        endpoint: "http://localhost:1234",
+        network: "mainnet",
+        publicClient: sepoliaClient,
+      });
+    }).toThrow("Network mismatch: expected mainnet (chain id 1), but got chain id 11155111.");
+  });
+
+  it("should throw an error for network mismatch on sepolia", () => {
+    const mainnetClient = createPublicClient({
       chain: mainnet,
       transport: http(PROVIDER_URI_MAINNET),
     });
 
-    const localNameguard = createClient({
-      // not a real endpoint, will error if used
-      endpoint: "http://localhost:1234",
-      publicClient
+    expect(() => {
+      createClient({
+        endpoint: "http://localhost:1234",
```

Comment:
Why are we passing an explicit `endpoint` param here? Why not leave it to the default?

Reply:
There's no point letting the client call nameguard since it should throw a different exception earlier. If it doesn't throw that exception then it would call nameguard and the test would fail (as expected) but that nameguard call would just slow down the test.